sal-common/src/main/java/org/ow2/proactive/sal/model/Hardware.java

```diff
@@ -251,30 +330,33 @@ public boolean equals(Object o) {
         return Objects.equals(this.id, hardware.id) && Objects.equals(this.name, hardware.name) &&
                Objects.equals(this.providerId, hardware.providerId) && Objects.equals(this.cores, hardware.cores) &&
                Objects.equals(this.ram, hardware.ram) && Objects.equals(this.disk, hardware.disk) &&
-               Objects.equals(this.fpga, hardware.fpga) && Objects.equals(this.location, hardware.location) &&
-               Objects.equals(this.state, hardware.state) && Objects.equals(this.owner, hardware.owner);
+               Objects.equals(this.fpga, hardware.fpga) && Objects.equals(this.gpu, hardware.gpu) && // Added GPU comparison
+               Objects.equals(this.cpuFrequency, hardware.cpuFrequency) && // Added CPU frequency comparison
```

Comment:
Remove comments.
Also consider using the @EqualsAndHashCode for all you equals and hascode implementations.
It is usually sufficient.